modules/platforms/cpp/ignite/odbc/sql_connection.cpp

```diff
@@ -119,9 +119,37 @@ void sql_connection::establish(const configuration &cfg) {
     IGNITE_ODBC_API_CALL(internal_establish(cfg));
 }
 
-void sql_connection::init_socket() {
-    if (!m_socket)
+sql_result sql_connection::init_socket() {
+    if (m_socket)
+        return sql_result::AI_SUCCESS;
+
+    if (m_config.get_ssl_mode().get_value() == ssl_mode_t::DISABLE) {
         m_socket = network::make_tcp_socket_client();
+
+        return sql_result::AI_SUCCESS;
+    }
+
+    try
+    {
+        network::ensure_ssl_loaded();
+    }
+    catch (const ignite_error &err)
+    {
+        LOG_MSG("Can not load OpenSSL library: " << err.what());
+
+        add_status_record("Can not load OpenSSL library (did you set OPENSSL_HOME environment variable?)");
```

Comment:
Should we check the actual value of `OPENSSL_HOME` env var and include it in the message?

E.g. `Failed to load OpenSSL library [OPENSSL_HOME='foobar']`

Reply:
Done.